pages/gateway/GatewayHandler.inc.php

```diff
@@ -175,19 +161,13 @@ public function clockss($args, $request)
 
             $prevYear = $nextYear = null;
             if (isset($year)) {
-                $result = $deprecatedIssueDao->retrieve(
-                    'SELECT MAX(year) AS max_year FROM issues WHERE journal_id = ? AND published = 1 AND year < ?',
-                    [$journal->getId(), $year]
-                );
-                $row = $result->current();
-                $prevYear = $row ? $row->max_year : null;
-
-                $result = $deprecatedIssueDao->retrieve(
-                    'SELECT MIN(year) AS min_year FROM issues WHERE journal_id = ? AND published = 1 AND year > ?',
-                    [$journal->getId(), $year]
-                );
-                $row = $result->current();
-                $nextYear = $row ? $row->min_year : null;
+                $key = $yearsIssuesPublished->search(function ($i) use ($year) {
+                    return $i === $year;
+                });
+                if (isset($key)) {
+                    $prevYear = $yearsIssuesPublished->get($key - 1);
```

Comment:
It looks like this is counting on database IDs being sequential, right? I'd rather not assume this -- e.g. an issue could be deleted.

Reply:
Yes, thanks for flagging this. That was the case. I'm switching it to use `Repo::issue()->getYearsIssuesPublished($journal->getId())->values()` when getting the collection at the start to reset the keys so this won't be a problem. Doing this for LOCKSS and for CLOCKSS.